addons/website_slides/models/slide_channel.py

```diff
@@ -534,21 +555,24 @@ def _get_categorized_slides(self, base_domain, order, force_void=True, limit=Fal
                 'total_slides': len(category_slides),
                 'slides': category_slides[(offset or 0):(limit + offset or len(category_slides))],
             })
-        # Then add uncategorized slides
+
+        # Add uncategorized slides. Note that uncategorized slides are inserted at the first position of the array
         uncategorized_slides = all_slides.filtered(lambda slide: not slide.category_id)
         if uncategorized_slides or force_void:
-            category_data.append({
-                'category': False, 'id': False,
-                'name': _('Uncategorized'), 'slug_name': _('Uncategorized'),
+            category_data.insert(0, {
+                'category': False,
+                'id': False,
+                'name': _('Uncategorized'),
+                'slug_name': _('Uncategorized'),
                 'total_slides': len(uncategorized_slides),
                 'slides': uncategorized_slides[(offset or 0):(offset + limit or len(uncategorized_slides))],
             })
+
         return category_data
 
-    def _resequence_slides(self, slide):
+    def _resequence_slides(self, slide, category_id=None):
         ids_to_resequence = self.slide_ids.ids
         index_of_added_slide = ids_to_resequence.index(slide.id)
-        category_id = slide.category_id.id
```

Comment:
Since the parameter is optional, you should make sure it works without it too.

So something like:

`new_category_id = category_id or slide.category_id.id`

And then use `new_category_id`.

Reply:
At this stage, the category_id set on the slide is not yet correct (depends on the sequence) so we cannot use it.
I think the only safe way to get its category_id, at this moment, is by using the one provided by the javascript to the controller.

Furthermore, not passing any category is nothing more than the "uncategorized slide" use case so I don't see how having category_id set as None could be problematic

Let me know if i'm being blind to something obvious.

Thanks for the review !